source/slang/slang-emit-wgsl.cpp

```diff
@@ -602,6 +602,21 @@ void WGSLSourceEmitter::emitVarKeywordImpl(IRType * type, IRInst* varDecl)
         m_writer->emit("storage, read");
         m_writer->emit(">");
     }
+    else if(varDecl->getOp() == kIROp_GlobalVar)
+    {
+        // Global ("module-scope") non-handle variables need to specify storage space
+
+        // https://www.w3.org/TR/WGSL/#var-decls
+        // "
+        // Variables in the private, storage, uniform, workgroup, and handle address
+        // spaces must only be declared in module scope, while variables in the function
+        // address space must only be declared in function scope. The address space must
+        // be specified for all address spaces except handle and function. The handle
+        // address space must not be specified. Specifying the function address space is
+        // optional.
+        // "
+        m_writer->emit("<private>");
```

Comment:
what about global groupshared var?

Reply:
I believe this case should be handled by this line, already:

https://github.com/shader-slang/slang/pull/5236/commits/26f6c47adefa746359b1f73050e47503af64901a#diff-9dc6d2e5a48c5258bcab72361355a660ee872bd88326ae87c28d6a26d5db9fe6L590

I think groupshared maps to WGSL's `workgroup` address space:
https://www.w3.org/TR/WGSL/#address-spaces-workgroup